### PR TITLE
create sitemapxml & robots

### PR DESCRIPTION
**Pull request by mickey530447**

### CODE REVIEW

File: controllers/Sitemap.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  try {
+    const PathDefault = 'https://canitrust.in/';
+    const TagDefault = 'https://canitrust.in/tag/';
+    const obj = {
+      urlset: {
+        '@xmlns': 'http://www.sitemaps.org/schemas/sitemap/0.9',
+        url: [],
+      },
+    };
+    const PathData = {
```
Comment (tuyenee) [inline controllers/Sitemap.js]:
```suggestion
    const PathData = await TestcaseService.getAllPageTestcase();
```

File: controllers/Sitemap.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      testcases: await TestcaseService.getAllPageTestcase(),
+    };
+    const homepageElement = {
+      loc: PathDefault,
+      lastmod: formatDate(
+        PathData.testcases[PathData.testcases.length - 1].date_created
```
Comment (tuyenee) [inline controllers/Sitemap.js]: testcases are not sorted, we must find the real last-modified testcase

File: controllers/Sitemap.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    for (const item of TagData.tags) {
+      const elementTag = {
+        loc: TagDefault.concat(item.tagText),
+        lastmod: formatDate('Mon July 1,2019'),
+        changefreq: 'Monthly',
+      };
+      obj.urlset.url.push(elementTag);
+    }
+    const feed = xmlbuilder.create(obj, { encoding: 'UTF-8' });
+    const tmp = `${feed.end({ pretty: true })}\n`;
```
Comment (tuyenee) [inline controllers/Sitemap.js]: adding \n may be unnecessary

File: controllers/Sitemap.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return res.status(200).send(tmp);
+  } catch (error) {
+    console.log(error);
+    return res.status(400).json(null);
+  }
+};
+
+function formatDate(date) {
+  const d = new Date(date);
+  let month = `${d.getMonth() + 1}`;
```
Comment (tuyenee) [inline controllers/Sitemap.js]:
```suggestion
  return d.toISOString().substring(0,10)
```

File: controllers/Sitemap.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      };
+      obj.urlset.url.push(elementPath);
+    }
+    const TagData = {
+      tags: await TagService.getAllTag(),
+    };
+    for (const item of TagData.tags) {
+      const elementTag = {
+        loc: TagDefault.concat(item.tagText),
+        lastmod: formatDate('Mon July 1,2019'),
```
Comment (tuyenee) [inline controllers/Sitemap.js]: Improvement: can we set `lastmod` of a tag with the date of lastest testcase of the tag?

File: controllers/Sitemap.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ * Licensed under the AGPLv3 License. See LICENSE.md in the project root for license information.
+ * ------------------------------------------------------------------------------------------------*/
+
+const xmlbuilder = require('xmlbuilder');
+const TagService = require('../service/TagService');
+const TestcaseService = require('../service/TestcaseService');
+
+module.exports.sitemapIndex = async (req, res) => {
+  try {
+    const PathDefault = 'https://canitrust.in/';
```
Comment (tuyenee) [inline controllers/Sitemap.js]: Please change https://canitrust.in to https://www.canitrust.in
also for the tag path

File: controllers/Sitemap.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable func-names */
+/* eslint-disable no-restricted-syntax */
```
Comment (trivien0) [inline controllers/Sitemap.js]: Should not disable ESLint restriction. Remove these lines and follow ESLint recommendation.

File: controllers/Sitemap.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      '@xmlns': 'http://www.sitemaps.org/schemas/sitemap/0.9',
+      url: [],
+    },
+  };
+  const tagModified = [];
+  try {
+    const testcaseData = await TestcaseService.getAllPageTestcase();
+    let testcaseLastModifiedDate = new Date('2018-01-01');
+    // Add testcase path in sitemap
+    for (const item of testcaseData) {
```
Comment (trivien0) [inline controllers/Sitemap.js]: Why you use for instead of foreach?

**Review by trivien0 — APPROVED** [review]
Approved!

**Pull request merged**